dotnet/src/SemanticKernel.Abstractions/Contents/ChatMessageContent.cs

```diff
@@ -18,20 +22,94 @@ public class ChatMessageContent : KernelContent
     public AuthorRole Role { get; set; }
 
     /// <summary>
-    /// Content of the message
+    /// A convenience property to get or set the text of the first item in the <see cref="Items" /> collection.
     /// </summary>
-    public string? Content { get; set; }
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public string? Content
+    {
+        get
+        {
+            if (this.Items.Count == 0)
+            {
+                return null;
+            }
+
+            if (this.Items[0] is TextContent textContent)
+            {
+                return textContent.Text;
+            }
+
+            throw new InvalidOperationException($"Cannot get the text content of the item of type {this.Items[0].GetType()}.");
+        }
+        set
+        {
+            if (value == null)
+            {
+                return;
+            }
+
+            if (this.Items.Count == 0)
+            {
+                this.Items.Add(new TextContent(
+                    text: value,
+                    modelId: this.ModelId,
+                    innerContent: this.InnerContent,
+                    encoding: this.Encoding,
+                    metadata: this.Metadata
+                ));
+                return;
+            }
+
+            if (this.Items[0] is TextContent textContent)
+            {
+                textContent.Text = value;
+                textContent.Encoding = this.Encoding;
+                return;
+            }
+
+            throw new InvalidOperationException($"Cannot set text content for the item of type {this.Items[0].GetType()}");
+        }
+    }
 
     /// <summary>
     /// Chat message content items
     /// </summary>
-    public ChatMessageContentItemCollection? Items { get; set; }
+    public ChatMessageContentItemCollection Items
+    {
+        get
+        {
+            return this._items ??
+                Interlocked.CompareExchange(ref this._items, new ChatMessageContentItemCollection(), null) ??
```

Comment:
Is this type intended to be read from concurrently? I would expect no, in which case this can just be `get => this._items ??= new ChatMessageContentItemCollection();`

Reply:
fixed.